Backend/src/database/services/user.service.ts

```diff
@@ -0,0 +1,82 @@
+import {IUser, userModel} from "../models/user";
+import {AddUserDTO, UpdateUserDTO} from "../dtos/user.dto";
+import {ObjectId} from "mongoose"
+
+/**
+ *  @class UserService
+ *
+ *  Provides useful methods to access the database and modify users,
+ *  which can be used by the route-controllers.
+ */
+export default class UserService {
+    /**
+     *  Gets all the users from the database.
+     *
+     *  @returns  allUsers
+     */
+    static async getAllUsers():
+      Promise<( IUser & { _id: ObjectId } )[]> {
+        return await userModel.find().exec();
+    }
+
+    /**
+     *  Search for a user with the given user id and return if found.
+     *
+     *  @param   user_id - the user id to search for
+     *  @param   includePassword - if true, the returned user includes the password field
+     *  @returns user - matched user to user_id or null
+     */
+    static async getUserById(user_id: (ObjectId | string), includePassword: boolean = false):
+      Promise<( IUser & { _id: ObjectId } | null )> {
+        return includePassword ?
+          await userModel.findById(user_id).select('+password').exec() :
+          await userModel.findById(user_id).exec();
+    }
+
+    /**
+     *  Search for a user with the given email and return if found.
+     *
+     *  @param   email - the email field to search for
+     *  @param   includePassword - if true, the returned user includes the password field
+     *  @returns user - matched user to email or null
+     */
+    static async getUserByEmail(email: string, includePassword: boolean = false):
+      Promise<( IUser & { _id: ObjectId } | null )> {
+        return includePassword ?
+          await userModel.findOne({email}).select('+password').exec() :
+          await userModel.findOne({email}).exec();
+    }
+
+    /**
+     *  Gets all unauthorized users.
+     *
+     *  @returns users - unauthorized users or null
+     */
+    static async getUnauthUsers():
+      Promise<( IUser & { _id: ObjectId } | null )[]> {
```

Comment:
Shouldn't return type be `Promise<(IUser & { _id: ObjectId })[] | null>` (an array containing users or no array at all, instead of an array containing users and null)?